contextualized/regression/lightning_modules.py

```diff
@@ -65,19 +85,134 @@ def __init__(
         self.model_regularizer = model_regularizer
         self.base_y_predictor = base_y_predictor
         self.base_param_predictor = base_param_predictor
-        self._build_metamodel(*args, **kwargs)
+
+        # different subclasses are passed different args
+        _NO_NUM_ARCH_CLASSES = ("NaiveContextualizedRegression",)
+        _HAVE_NUM_ARCH_CLASSES = (
+            "ContextualizedRegression",
+            "MultitaskContextualizedRegression",
+            "ContextualizedUnivariateRegression",
+        )
+        _TASK_SPLIT_CLASSES = (
+            "TasksplitContextualizedRegression",
+            "TasksplitContextualizedUnivariateRegression",
+        )
+        if self.__class__.__name__ in _NO_NUM_ARCH_CLASSES:
+            self._build_metamodel(
+                context_dim,
+                x_dim,
+                y_dim,
+                univariate=univariate,
+                encoder_type=encoder_type,
+                width=width,
+                layers=layers,
+                link_fn=link_fn,
+            )
```

Comment:
This will fail silently if the user specifies any of the TaskSplitMetamodel args when they are using a non-tasksplit class. Same for other args like num_archetypes if the user using NaiveContextualizedRegression. These should fail at init.